src/main/java/com/example/moviemanager/service/MovieService.java

```diff
@@ -36,17 +64,62 @@ public void deleteById(UUID id) {
         }
     }
 
-    public Movie save(Movie movie) {
-        return movieRepository.save(movie);
+    public ExistingMovieDetails save(CreateMovieInfo createMovieInfo) {
+        MovieDetailsFromApi movieDetailsFromApi = getDetailsFromMovieDatabase(createMovieInfo);
+        Movie createdMovie = movieRepository.save(buildMovie(createMovieInfo, movieDetailsFromApi));
+
+        return ExistingMovieDetails.fromEntity(createdMovie);
     }
 
-    public Movie update(UUID id, Movie movie) {
+    public ExistingMovieDetails update(UUID id, UpdateMovieInfo updateMovieInfo) {
         Optional<Movie> tempMovie = movieRepository.findById(id);
         if (tempMovie.isEmpty()) {
             throw new MovieNotFoundException("Update failed. Could not find movie by id - " + id);
         }
-        movie.setId(id);
 
-        return movieRepository.save(movie);
+        Movie movieWithNewDetails = updateMovieInfo.toEntity();
+        movieWithNewDetails.setId(id);
+        Movie updatedMovie = movieRepository.save(movieWithNewDetails);
+
+        return ExistingMovieDetails.fromEntity(updatedMovie);
+    }
+
+    private MovieDetailsFromApi getDetailsFromMovieDatabase(CreateMovieInfo createMovieInfo) {
+        String uri = buildUri(createMovieInfo);
+        ApiResponse apiResponse = getResponseFromMovieDatabase(uri);
+
+        return apiResponse.results().get(0);
+    }
+
+    private Movie buildMovie(CreateMovieInfo saveMovieRequest, MovieDetailsFromApi movieDetailsFromApi) {
+        return new Movie(
+                saveMovieRequest.title().toUpperCase(),
+                movieDetailsFromApi.overview(),
+                saveMovieRequest.releaseYear(),
+                movieDetailsFromApi.vote_average()
+        );
+    }
+
+    private String buildUri(CreateMovieInfo saveMovieRequest) {
+        String movieType = saveMovieRequest.type();
+        if (movieType.equalsIgnoreCase(MovieType.MOVIE.toString())) {
+            return MOVIES_URI + saveMovieRequest.title() + MOVIE_RELEASE_YEAR + saveMovieRequest.releaseYear();
+        }
+        return TV_SHOWS_URI + saveMovieRequest.title() + TV_SHOW_RELEASE_YEAR + saveMovieRequest.releaseYear();
+    }
+
+    private ApiResponse getResponseFromMovieDatabase(String uri) {
+        try {
+            ApiResponse apiResponse = objectMapper.readValue(
+                    new RestTemplate().getForObject(uri, String.class),
+                    ApiResponse.class
+            );
+            if(apiResponse.results().size() != 1) {
+                throw new UniqueMovieNotFoundException("Can`t find unique movie according to your details.");
+            }
+            return apiResponse;
+        } catch (JsonProcessingException e) {
+            throw new FailedToRetrieveDataFromDatabaseException("Failed to retrieve data from database.");
```

Comment:
JsonProcessingException means that you wasn't able to map response to DTO I think, so your custom exception is a bit misleading